digital_asset_client.go

```diff
@@ -0,0 +1,553 @@
+	// Serialize arguments
+	collectionBytes, err := bcs.SerializeString(options.CollectionName)
+	if err != nil {
+		return CreateErrorMessage("collection", err)
+	}
+
+	descriptionBytes, err := bcs.SerializeString(options.TokenDescription)
+	if err != nil {
+		return CreateErrorMessage("description", err)
+	}
+
+	nameBytes, err := bcs.SerializeString(options.TokenName)
+	if err != nil {
+		return CreateErrorMessage("name", err)
+	}
+
+	uriBytes, err := bcs.SerializeString(options.TokenURI)
+	if err != nil {
+		return CreateErrorMessage("URI", err)
+	}
+
+	// Serialize property keys
+	propertyKeysBytes, err := SerializeVectorString(options.PropertyKeys)
+	if err != nil {
+		return CreateErrorMessage("property_keys", err)
+	}
+
+	// Serialize property types
+	propertyTypesBytes, err := SerializeVectorString(options.PropertyTypes)
+	if err != nil {
+		return CreateErrorMessage("property_types", err)
+	}
+
+	// Serialize property values (as bytes)
+	propertyValuesBytes, err := SerializePropertyValues(options.PropertyValues, options.PropertyTypes)
+	if err != nil {
+		return CreateErrorMessage("property_values", err)
+	}
+
+	// Build the transaction
+	var functionName string
+	var args [][]byte
+
+	if options.IsSoulBound {
+		functionName, args, err = mintSoulboundArgs(recipient, collectionBytes, descriptionBytes, nameBytes, uriBytes, propertyKeysBytes, propertyTypesBytes, propertyValuesBytes)
+		if err != nil {
+			return "", err
+		}
+	} else {
+		functionName, args = mintArgs(collectionBytes, descriptionBytes, nameBytes, uriBytes, propertyKeysBytes, propertyTypesBytes, propertyValuesBytes)
+	}
+
+	rawTxn, err := dac.client.BuildTransaction(creator.AccountAddress(), TransactionPayload{
+		Payload: &EntryFunction{
+			Module: ModuleId{
+				Address: AccountFour, // 0x4
+				Name:    "aptos_token",
+			},
+			Function: functionName,
+			ArgTypes: []TypeTag{},
+			Args:     args,
+		},
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to build transaction: %w", err)
+	}
+
+	// Sign and submit the transaction
+	signedTxn, err := rawTxn.SignedTransaction(creator)
+	if err != nil {
+		return "", fmt.Errorf("failed to sign transaction: %w", err)
+	}
+
+	submitResult, err := dac.client.SubmitTransaction(signedTxn)
+	if err != nil {
+		return "", fmt.Errorf("failed to submit transaction: %w", err)
+	}
+
+	return submitResult.Hash, nil
+}
+
+func mintSoulboundArgs(recipient AccountAddress, collectionBytes []byte, descriptionBytes []byte, nameBytes []byte, uriBytes []byte, propertyKeysBytes []byte, propertyTypesBytes []byte, propertyValuesBytes []byte) (string, [][]byte, error) {
+	functionName := "mint_soul_bound"
+
+	// Serialize the soul_bound_to address
+	recipientBytes, err := bcs.Serialize(&recipient)
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to serialize recipient: %w", err)
+	}
+
+	// mint_soul_bound requires the recipient address as the last argument
+	args := [][]byte{
```

Comment:
I would probably take in the types, and convert them instead in the args functions, so you don't have to pass in a large list of bytes (where you might get one out of order)